### DISCUSSION TITLE
[V5 ] Get current user in model hook

### DISCUSSION THREAD

**Discussion by lowbits** [adonisjs/core#2474] [Help]
Hello, 
so i have an application where users can buy courses. 
When a logged in user tries to get all courses, i want to add the latest answered lesson to the course model. 
I tried many things, but nothing really works out.
Another approach would be to fetch the latestActiveLesson by an extra API-endpoint, but then i would have n^2 requests.

This is my ugly code so far:

My problem in this approach is, that i have no access to the current signed in user `getLatestActiveLessonFromUser()`

```javascript 
public set setActiveLesson(activeLesson: LatestActiveLesson) {
    this.latestActiveLesson = activeLesson
  }

  @computed()
  public latestActiveLesson: LatestActiveLesson

  @afterFetch()
  public static async addLatestLesson(courses: Course[]) {
    const activeLessonsPromise = courses.map(async (course) => {
      await course.getLatestActiveLessonFromUser()
    })

    const activeLessons = await Promise.all(activeLessonsPromise)

    courses.forEach((course) => {
      const lesson = activeLessons.find((lesson) => lesson.id === course.id)
      course.setActiveLesson(lesson)
    })
  }
  ```
  
  My models
```javascript
Course
 @hasMany(() => Chapter)
  public chapters: HasMany<typeof Chapter>
  
Chapter 
@hasMany(() => Lesson)
  public lessons: HasMany<typeof Lesson>
  
Lesson
@hasMany(() => LatestActiveLesson)
  public latestActiveLesson: HasMany<typeof LatestActiveLesson>
```
  
  I know this piece of code is not that pretty. If you have any other ideas how to accomplish my task, i would by glad to hear.

**Comment by RomainLanz** [comment]
Hey @lowbits! 👋 

Why not preload your relation from your controller? That seems easier and better in term of performance than what you are doing.

**Reply by lowbits** [reply]
The problem with preloading is, that you add all chapters and lessons to the course object, but you only need course info and latestActiveLesson. 
Maybe related does the trick.

**Reply by 3rownz** [reply]
Its hard to tell what your database and output looks like, but if activeLesson and lesson have similar data, then add a is_active boolean to the database. Im not sure what `getLatestActiveLessonFromUser()` does, but if a user is related to the course, this relalationship should be added (then you can compare against auth.user).  Like RomainLanz said use a preload like this could help:
```
const course = await Course
  .query()
  .select('*')
  .where('user_id', auth.user.id)
  .preload('lesson', (lessonQuery) => {
    lessonQuery
       .where('is_active', true)  
  })
```
